### PR TITLE
Add hidden form field

### PR DESCRIPTION
**Pull request by Soare-Robert-Daniel**
Closes #.

### Summary

Add a field that is hidden from the client. Its function is to collect data from URLs to save metadata along with the submission.

URL example: `http://localhost:10007/496/form-hidden-field/?test=123`

#### Editor

image
#### Inspector

image

#### Frontend

image

#### Submission

image

----

### Test instructions

---- 

### Checklist before the final review

- [x] Included E2E or unit tests for the changes in this PR.
- [x] Visual elements are not affected by independent changes.
- [x] It is at least compatible with the [minimum WordPress version](https://wordpress.org/plugins/otter-blocks/).
- [x] It loads additional script in frontend only if it is required.
- [x] Does not impact the [Core Web Vitals](https://web.dev/vitals/).
- [x] In case of deprecation, old blocks are safely migrated.
- [x] It is usable in Widgets and FSE.
- [x] Copy/Paste is working if the attributes are modified.
- [x] PR is following [the best practices]()

### CODE REVIEW

File: src/pro/blocks/form-hidden-field/edit.js [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+						])
+					}
+				</style>
+				<label
+					htmlFor={ attributes.id }
+					className="otter-form-input-label"
+				>
+					<span className="o-hidden-field-mark">
+						{ __( 'Hidden Field', 'otter-blocks' ) }
+					</span>
```
Comment (HardeepAsrani) [inline src/pro/blocks/form-hidden-field/edit.js:63]: As talked on Slack, we don't need to show this. We can simply show it like a normal field when Form block is selected and not show the block if isSelected is false.

**Review by HardeepAsrani — CHANGES_REQUESTED** [review]
Apart from mentioned, few things:

- The Block doesn't appear in the select list, nor the select list appear inside the block
Screenshot 2023-06-23 at 8 56 06 AM

- If you click on Back to the Form button, it generates console errors.

File: src/blocks/blocks/form/hidden-field/inspector.js [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+		<InspectorControls>
+			<PanelBody
+				title={ __( 'Field Settings', 'otter-blocks' ) }
+			>
+				<Button
+					isSecondary
+					variant="secondary"
+					onClick={ () => selectForm?.() }
+				>
+					{ __( 'Back to the Form', 'otter-blocks' ) }
```
Comment (HardeepAsrani) [inline src/blocks/blocks/form/hidden-field/inspector.js:54]: This still doesn't work for me.

**Review by HardeepAsrani — CHANGES_REQUESTED** [review]
Just a minor change and it should be good to go for QA.

**Review by HardeepAsrani — APPROVED** [review]

**Comment by rodica-andronache** [comment]
@Soare-Robert-Daniel 

- [ ] when I activated Otter Pro, I got a fatal error https://vertis.d.pr/i/ewWgrU https://vertis.d.pr/i/1WZtgA
- [ ] in the free version, unlike the File field, which has its options disabled https://vertis.d.pr/i/iHWcCw, for the Hidden field the options can be edited sometimes https://vertis.d.pr/v/GcwqVy 
- [ ] in the free version, sometimes, after choosing the Hidden field you can't change that anymore, to other types of fields https://vertis.d.pr/v/cy58j5

**Comment by Soare-Robert-Daniel** [comment]
> * [ ]  when I activated Otter Pro, I got a fatal error https://vertis.d.pr/i/ewWgrU https://vertis.d.pr/i/1WZtgA

I could not reproduce the fatal error.

**Comment by rodica-andronache** [comment]
@Soare-Robert-Daniel I've tried again, and the fatal error is still happening for me. Maybe this video could help to see the exact steps I take ( the fatal error occurs after activating the license ) https://www.loom.com/share/62a6e77daaee48f09755eac2f562b008?sid=7619d799-4994-4b09-ba58-fc0541fc90e1
Let me know if there is something else I can share to help out. Thank you!

**Comment by HardeepAsrani** [comment]
@Soare-Robert-Daniel I think it's because of the wrong name of the file: https://github.com/Codeinwp/otter-blocks/blob/feat/form-hidden-field/plugins/otter-pro/inc/render/class-form-hidden-block.pnp.php

It's `class-form-hidden-block.pnp.php` at this moment.

**Comment by Soare-Robert-Daniel** [comment]
@HardeepAsrani thanks for info 🙏 . It seams that local tools did not find anything strange with that odd name.

Testing instance 

```
 Admin area URL: https://fiveact.s4-tastewp.com/wp-admin 
 Username: robertino 
 Password: pZ8pJCocVZI 
```

**Comment by rodica-andronache** [comment]
@Soare-Robert-Daniel everything's working well 🚀

**Pull request merged**